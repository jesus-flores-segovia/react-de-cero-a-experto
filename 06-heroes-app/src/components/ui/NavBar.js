import React, { useContext } from 'react'
import { NavLink, useHistory } from 'react-router-dom'
import { logout } from '../../actions/authActions';
import { AuthContext } from '../../auth/AuthContext'

export const Navbar = () => {

  const {user, dispatch} = useContext(AuthContext);
  const history = useHistory();

  const handleLogout = () => {
    dispatch(logout());
    history.replace("/login");
  };

  return (
    <nav className="navbar navbar-expand-lg navbar-light bg-light">
      <div className="container-fluid">
        <a className="navbar-brand ms-2" href="/">
          <img className="me-2" src="/assets/icons/icons8-hulk-48.png" width="30" height="30" alt="heroes"/>
          Heroes
        </a>
        <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarText" aria-controls="navbarText" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon"></span>
        </button>
        <div className="collapse navbar-collapse" id="navbarText">
          <ul className="navbar-nav me-auto mb-2 mb-lg-0">
            <NavLink 
                activeClassName="active"
                className="nav-link" 
                exact
                to="/marvel"
            >
                Marvel Comics
            </NavLink>    

            <NavLink 
                activeClassName="active"
                className="nav-link" 
                exact
                to="/dc"
            >
                DC Comics
            </NavLink>

            <NavLink 
                activeClassName="active"
                className="nav-link" 
                exact
                to="/search"
            >
                Search
            </NavLink>
          </ul>
      </div>
      <div className="collapse navbar-collapse justify-content-end" id="navbarText">
        <ul className="navbar-nav mb-2 mb-lg-0 align-items-center">
        {user.name &&
          <span className="nav-item text-info me-3">You're logged as {user.name}</span>
        }
          <button type="button" className="btn btn-danger" style={{color:"white", textDecoration:"none"}} onClick={handleLogout}>
            Logout
          </button>
        </ul>
      </div>
    </div>
  </nav>
  )
}