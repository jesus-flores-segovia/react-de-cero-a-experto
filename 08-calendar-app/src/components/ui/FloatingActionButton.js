import React from "react";

export const FloatingActionButton = (props) => {
  return (
    <button className={props.className} onClick={props.onClick}>
      {props.children}
    </button>
  );
};
