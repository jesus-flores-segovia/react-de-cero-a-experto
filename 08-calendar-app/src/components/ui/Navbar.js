import React from "react";

export const Navbar = () => {
  return (
    <div className="navbar navbar-dark bg-dark mb-4">
      <span className="navbar-brand">Jes√∫s</span>
      <button className="btn btn-danger">
        <i className="fas fa-sign-out-alt" />
        <span> Logout</span>
      </button>
    </div>
  );
};
