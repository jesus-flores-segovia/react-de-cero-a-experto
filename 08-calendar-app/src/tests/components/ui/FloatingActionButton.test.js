import { mount } from "enzyme";
import { FloatingActionButton } from "../../../components/ui/FloatingActionButton";

const deleteEvent = jest.fn();

const wrapper = mount(
  <FloatingActionButton
    className="btn btn-danger fab-danger"
    onClick={deleteEvent}
  >
    <i className="fas fa-trash" />
    <span> Delete event</span>
  </FloatingActionButton>
);

describe("Tests inside file 'FloatingActionButton.test.js'", () => {
  test("Must be shown correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Component must fire the configured event", () => {
    wrapper.find("button").prop("onClick")();
    expect(deleteEvent).toHaveBeenCalledTimes(1);
  });
});
