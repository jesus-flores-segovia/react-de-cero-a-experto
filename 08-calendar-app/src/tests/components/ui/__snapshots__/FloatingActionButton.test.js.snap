// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests inside file 'FloatingActionButton.test.js' Must be shown correctly 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[MockFunction]}
>
  <i
    className="fas fa-trash"
  />
  <span>
     Delete event
  </span>
</button>
`;
